### PR TITLE
add deserialize_ignore_any

### PR DESCRIPTION
**Pull request by lovasoa**
Fixes https://github.com/jonasbb/serde_with/issues/250

### CODE REVIEW

File: src/rust.rs [hunk 1]
```diff
@@ -1802,3 +1802,23 @@ pub mod default_on_null {
         value.serialize(serializer)
     }
 }
+
+/// Deserialize any value, ignore it, and return the default value for the type being deserialized
```
Comment (jonasbb) [inline src/rust.rs:1806]: Could you please extend the description with a note that this only works for enums with data.
And maybe also the part about simulating an other-like variant in untagged enums. Basically the info in my last comment in #250.

**Review by jonasbb — CHANGES_REQUESTED** [review]
Thanks for the contribution. Could you please extend the description a bit.

If you want a test case would be nice to have but I am happy to merge it already and add the test case later.

File: src/rust.rs (hunk 1 above)
Comment (lovasoa) [reply src/rust.rs]: I added a description giving more context about when and why the function should be used.

**Comment by jonasbb** [comment]
Thanks for updating the description. Once CI passes you can merge the PR by writing `bors r+` in a comment.

bors delegate=lovasoa

**Pull request merged**